backend/src/main/scala/cromwell/backend/standard/callcaching/BlacklistCache.scala

```diff
@@ -18,7 +19,23 @@ case class BlacklistCache(config: CacheConfig) {
       build[String, java.lang.Boolean](falseLoader)
   }
 
-  def isBlacklisted(bucket: String): Boolean = cache.get(bucket)
+  val hitCache = {
+    // Queries to the hit blacklist cache return false by default (i.e. not blacklisted).
+    val falseLoader = new CacheLoader[CallCachingEntryId, java.lang.Boolean]() {
+      override def load(key: CallCachingEntryId): java.lang.Boolean = false
+    }
+
+    CacheBuilder.
+      newBuilder().
+      concurrencyLevel(config.concurrency).
+      maximumSize(config.size).
+      expireAfterWrite(config.ttl.length, config.ttl.unit).
+      build[CallCachingEntryId, java.lang.Boolean](falseLoader)
+  }
+
+  def isBlacklisted(hit: CallCachingEntryId, bucket: String): Boolean = hitCache.get(hit) || bucketCache.get(bucket)
```

Comment:
Could this return a `BlacklistStatus` object like `NotBlacklisted`, `BlacklistedHit`, `BlacklistedBucket`, `BlacklistedHitAndBucket` so that our metrics could record them separately (as well as in aggregate)?

Reply:
I agree all of this should be metric'd. I'll start working on that and see where it goes. 🙂